File: springboot/src/main/java/com/xeon/springboot/controller/UserController.java
package com.xeon.springboot.controller;


import com.xeon.springboot.dao.UserMappper;
import com.xeon.springboot.dao.data.User;
import com.xeon.springboot.service.UserService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.ResponseBody;

import java.util.List;

@Controller
@RequestMapping("/user")
public class UserController {

    @Autowired
    private UserService userService;


    @RequestMapping(value = "/listUsers", method = {RequestMethod.GET})
    @ResponseBody
    public List<User> listUsers() {
        final List<User> users = userService.listUsers();

        System.out.println("users = " + users);


        return users;

    }


    @RequestMapping("/getUser")
    @ResponseBody
    public User getUserById(int id){
        return userService.getUser(id);
    }

}

File: kotlin-springboot-kts/src/main/kotlin/com/yzq/kotlinspringbootkts/controller/UserController.kt
package com.yzq.kotlinspringbootkts.controller

import com.yzq.kotlinspringbootkts.data.User
import com.yzq.kotlinspringbootkts.service.UserService
import com.yzq.kotlinspringbootkts.dto.CreateUserRequest
import com.yzq.kotlinspringbootkts.dto.UpdateUserRequest
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/user")
class UserController {


    @Autowired
    private lateinit var userService: UserService


    @GetMapping("/list")
    fun list(): List<User> {
        return userService.list()
    }

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Long): User? {
        return userService.getById(id)
    }

    @PostMapping
    fun create(@RequestBody req: CreateUserRequest): Int {
        val user = User(
            id = 0L,
            name = req.name,
            age = req.age,
            gender = req.gender,
            createTime = null,
            updateTime = null,
            enable = req.enable
        )
        return userService.create(user)
    }

    @PutMapping("/{id}")
    fun update(
        @PathVariable("id") id: Long,
        @RequestBody req: UpdateUserRequest
    ): Int {
        val user = User(
            id = id,
            name = req.name,
            age = req.age,
            gender = req.gender,
            createTime = null,
            updateTime = null,
            enable = req.enable
        )
        return userService.updateById(user)
    }

    @DeleteMapping("/{id}")
    fun delete(@PathVariable("id") id: Long): Int {
        return userService.deleteById(id)
    }


}
